homeassistant/components/binary_sensor/fibaro.py

```diff
@@ -0,0 +1,72 @@
+"""
+Support for Fibaro binary sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.fibaro/
+"""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, ENTITY_ID_FORMAT)
+from homeassistant.components.fibaro import (
+    FIBARO_CONTROLLER, FIBARO_DEVICES, FibaroDevice)
+
+DEPENDENCIES = ['fibaro']
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_TYPES = {
+    'com.fibaro.doorSensor':   ['Door', 'mdi:window-open', 'door'],
+    'com.fibaro.windowSensor': ['Window', 'mdi:window-open', 'window'],
+    'com.fibaro.smokeSensor': ['Smoke', 'mdi:smoking', 'smoke'],
+    'com.fibaro.FGMS001': ['Motion', 'mdi:run', 'motion'],
+    'com.fibaro.heatDetector': ['Heat', 'mdi:fire', 'heat'],
+}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Perform the setup for Fibaro controller devices."""
+    add_entities(
+        [FibaroBinarySensor(device, hass.data[FIBARO_CONTROLLER])
+         for device in hass.data[FIBARO_DEVICES]['binary_sensor']], True)
+
+
+class FibaroBinarySensor(FibaroDevice, BinarySensorDevice):
+    """Representation of a Fibaro Binary Sensor."""
+
+    _icon = None
+    _device_class = None
```

Comment:
Why not move these inside `__init__` ?

Reply:
You mean the declarations and initializations? I'm coming from c/c++ and somehow it feels nicer to know what variables the functions of a class use, takes a bit of the guesswork out. Not sure if it's pythonic :-) If you say it should be moved, I'm happy to move it, tho.

Reply:
Python does not use explicit declarations, what you have here are class variables (static members in C++). It is not exactly wrong because your instance variables will override them ... but it looks odd.

The guesswork is largely avoided by initializing all instance variables in the `__init__` method.

Reply:
I see. Thanks. I'll move it in a future PR, if that's okay

Reply:
I would prefer that we do this now so we have the pattern right when the remaining device types arrive.

It seems like we should always end up with a not-None `stype`? In that case these initializers can just be removed.

Reply:
stype can be None. I couldn't find a documentation of all the various device types and basetypes Fibaro HC supports and can report, so these are more like heuristics, trying to match it with known types, but at the same type keeping any other type support. Even if their icon might not be specialized, but at least all binary sensors would be supported this way.
I did some cleanup, to only initialize with None if needed.

Reply:
Right, I forgot that devices are also added based on capabilities.